### ISSUE TITLE
Passing extra arguments down to the tools e4k calls

### ISSUE THREAD

**Issue by spouliot** [mono/Embeddinator-4000#334]
e4k can (optionally) compile the code it generates, e.g. using `clang`. It would be useful for the user to be able to add extra files (#130) and also other arguments to compile (and to link). Otherwise the caller has to re-implement parts of e4k logic to build his custom project.

The same happens when e4k calls SDK packaging tools , e.g. `mtouch` or `mmp`. We do not want to add every possible options to e4k (e.g. `--xml` in #301), there's just too many of them.

What we can add are generic options, like

* `--compiler-extra-args="... unvalidated options given to the compiler ..."`;
* `--linker-extra-args="... unvalidated options given to the native linker ..."`;
* `--sdk-extra-args="... unvalidated options given to the application SDK tool ..."`;

That last name is a placeholder for a better suggestion :)

**Comment by dalexsoto** [comment]
Yeah kinda like sharpie does for clang with the -c option, this is a great idea, I think I like more your format

`--compiler-extra-args="... unvalidated options given to the compiler ...` than the `-c` way in sharpie
